include/zephyr/net/net_if.h

```diff
@@ -2643,6 +2643,21 @@ void net_if_unregister_link_cb(struct net_if_link_cb *link);
 void net_if_call_link_cb(struct net_if *iface, struct net_linkaddr *lladdr,
 			 int status);
 
+/**
+ * @brief Type of checksum for which support in the interface will be queried.
+ */
+enum net_if_checksum_protocol_type{
```

Comment:
Actually, looking at the code this enum name looks a bit lengthy ,maybe just call this `net_if_checksum_type` and `NET_IF_CHECKSUM_IPV4` etc...

Reply:
OK, I will change it accordingly.